src/app/component/settings-panel/settings-panel.component.ts

```diff
@@ -213,7 +219,7 @@ export class SettingsPanelComponent {
     }
   }
 
-  numRegex = '^-?[0-9]+(.[0-9]{0,10})?$';
+  numRegex = '^-?[0-9]+(\\.[0-9]{0,10})?\\s*[a-zA-Z]*$';
```

Comment:
I don't think we should use the regex for the input field with a value and unit. Instead the current behavior for the other combo input fields (where it's not just a number) is to revert back to the original number. (Try this for the edit panel input fields)